### PR TITLE
Core delete project route

### PR DESCRIPTION
**Pull request by PopDaph**
New route to delete data from a given project in Core.
We don't delete if there are datasources for this project. 

Required for a Poké feature to delete all data from a workspace. 

🦀

### CODE REVIEW

**Review by spolu — COMMENTED** [review]
Left a few comments that need addressing but otherwise structure LGTM :+1: Feel free to merge once addressed :+1:

File: core/src/stores/postgres.rs
Comment (PopDaph) [reply core/src/stores/postgres.rs]: OK so problem is we need to delete the `runs_joins` before we delete the _block_executions_. I tried handling it with a where in with keeping track of the ids. Not sure if that's better then the way I did it for the `datasets_points`. I'd love your opinion on bulding a query with a where in or multiple where id =. I'll then add a commit to handle both use case with the same method 🙏🏻

Comment (PopDaph) [reply core/src/stores/postgres.rs]: I made an attempt with a function so that we are in SQL, seems to work like that!

**Review by spolu — DISMISSED** [review]

File: core/src/stores/postgres.rs
Comment (spolu) [inline core/src/stores/postgres.rs:82]: Let's move to store.rs as CREATE IF NOT EXISTS :pray:

File: core/src/stores/postgres.rs
Comment (spolu) [inline core/src/stores/postgres.rs:114]: Let's move to store.rs as CREATE IF NOT EXISTS :pray:

File: core/src/stores/postgres.rs
Comment (spolu) [reply core/src/stores/postgres.rs:82]: (feel free to merge directly after of course!)

Comment (PopDaph) [reply core/src/stores/postgres.rs]: There is no "create if not exists" for functions. 🙃 

I moved them to store, so now they will be re-created at each deploy. 
I think it's better then managing them with a one time migration because they need to stay up to date at all time and engineers will look at this file to manage any change in the db. I'll add a commit to change it if you're against it 😬

File: core/src/stores/postgres.rs
Comment (PopDaph) [reply core/src/stores/postgres.rs]: Answered in the other comment, it does not exist 😢

File: core/src/stores/postgres.rs
Comment (fontanierh) [inline core/src/stores/postgres.rs:57]: I think this is just equivalent to `c.execute(..).await?;` no ? Don't think you need a match

Comment (PopDaph) [reply core/src/stores/postgres.rs:57]: For this part I just did the same as for the tables and indexes TBH 🙈

File: core/src/stores/postgres.rs
Comment (spolu) [reply core/src/stores/postgres.rs]: Works for me. We just want to make sure concurrent deploys don't crash the thing.

File: core/src/stores/postgres.rs
Comment (spolu) [reply core/src/stores/postgres.rs]: I guess it may, but I presume it'll just be fine

**Pull request merged**
